src/System.Text.Json/tests/JsonArrayTests.cs

```diff
@@ -0,0 +1,344 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonArrayTests
+    {
+        [Fact]
+        public static void TestAdd()
+        {
+            var jsonArray = new JsonArray();
+            int idx = 0;
+
+            jsonArray.Add("value");
+            Assert.Equal("value", (JsonString)jsonArray[idx++]);
+            Assert.True(jsonArray.Contains("value"));
+
+            jsonArray.Add(true);
+            Assert.True(((JsonBoolean)jsonArray[idx++]).Value);
+            Assert.True(jsonArray.Contains(true));
+
+            jsonArray.Add(byte.MaxValue);
+            Assert.Equal(byte.MaxValue, ((JsonNumber)jsonArray[idx++]).GetByte());
+            Assert.True(jsonArray.Contains(byte.MaxValue));
+
+            jsonArray.Add(short.MaxValue);
+            Assert.Equal(short.MaxValue, ((JsonNumber)jsonArray[idx++]).GetInt16());
+            Assert.True(jsonArray.Contains(short.MaxValue));
+
+            jsonArray.Add(int.MaxValue);
+            Assert.Equal(int.MaxValue, ((JsonNumber)jsonArray[idx++]).GetInt32());
+            Assert.True(jsonArray.Contains(int.MaxValue));
+
+            jsonArray.Add(long.MaxValue);
+            Assert.Equal(long.MaxValue, ((JsonNumber)jsonArray[idx++]).GetInt64());
+            Assert.True(jsonArray.Contains(long.MaxValue));
+
+            jsonArray.Add(3.14f);
+            Assert.Equal(3.14f, ((JsonNumber)jsonArray[idx++]).GetSingle());
+
+            jsonArray.Add(3.14);
+            Assert.Equal(3.14, ((JsonNumber)jsonArray[idx++]).GetDouble());
+
+            jsonArray.Add(sbyte.MaxValue);
+            Assert.Equal(sbyte.MaxValue, ((JsonNumber)jsonArray[idx++]).GetSByte());
+
+            jsonArray.Add(ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt16());
+
+            jsonArray.Add(uint.MaxValue);
+            Assert.Equal(uint.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt32());
+
+            jsonArray.Add(ulong.MaxValue);
+            Assert.Equal(ulong.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt64());
+
+            jsonArray.Add(decimal.One);
+            Assert.Equal(decimal.One, ((JsonNumber)jsonArray[idx++]).GetDecimal());
+        }
+
+        [Fact]
+        public static void TestInsert()
+        {
+            var jsonArray = new JsonArray();
+
+            jsonArray.Insert(0, "value");
+            Assert.Equal("value", (JsonString)jsonArray[0]);
+
+            jsonArray.Insert(0, true);
+            Assert.True(((JsonBoolean)jsonArray[0]).Value);
+
+            jsonArray.Insert(0, byte.MaxValue);
```

Comment:
All of these insert at 0.  There should also be a set that inserts at the end. (Yeah, the first one is both 0 and the end, but this is "show that Length is special, not that 0 is special")